### PR TITLE
[Application] Enforce minimum stake when staking

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

- Adds minimum stake validation to the application stake message handler (i.e. total stake must be >= minimum stake).
- Updates error returns in the same handler to ensure all error paths return appropriate gRPC status errors.
- Replaces some warn and error level logs with info level, which I believe is more appropriate (until we have a practical debug level, at which point they should become debug logs).

## Dependencies

- #809
- #843 
- #844 
- #845

## Dependents
 
- #848 
- #849
- #850
- #857
- #852 
- #861 
- #851 
- #863 

## Issue

- #612

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [x] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [x] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: x/application/keeper/msg_server_stake_application_test.go [hunk 1]
```diff
@@ -42,14 +44,19 @@ func TestMsgServer_StakeApplication_SuccessfulCreateAndUpdate(t *testing.T) {
 	foundApp, isAppFound := k.GetApplication(ctx, appAddr)
 	require.True(t, isAppFound)
 	require.Equal(t, appAddr, foundApp.Address)
-	require.Equal(t, int64(100), foundApp.Stake.Amount.Int64())
+	require.Truef(t,
+		initialStake.Amount.Equal(foundApp.Stake.Amount),
+		"expected %d, got %d",
+		initialStake.Amount.Int64(), foundApp.Stake.Amount.Int64(),
+	)
```
Comment (bryanchriswhite) [inline x/application/keeper/msg_server_stake_application_test.go:51]:
```suggestion
	require.Equal(t, initialStake.Amount.Int64(), foundApp.Stake.Amount.Int64())
```

File: x/application/keeper/msg_server_stake_application_test.go [hunk 2]
```diff
@@ -61,7 +68,11 @@ func TestMsgServer_StakeApplication_SuccessfulCreateAndUpdate(t *testing.T) {
 	require.NoError(t, err)
 	foundApp, isAppFound = k.GetApplication(ctx, appAddr)
 	require.True(t, isAppFound)
-	require.Equal(t, int64(200), foundApp.Stake.Amount.Int64())
+	require.Truef(t,
+		upStake.Amount.Equal(foundApp.Stake.Amount),
+		"expected %d, got %d",
+		upStake.Amount.Int64(), foundApp.Stake.Amount.Int64(),
+	)
```
Comment (bryanchriswhite) [inline x/application/keeper/msg_server_stake_application_test.go:75]:
```suggestion
	require.Equal(t, upStake.Amount.Int64(), foundApp.Stake.Amount.Int64())
```

File: x/application/keeper/msg_server_transfer_application_stake_test.go [hunk 3]
```diff
@@ -130,7 +128,11 @@ func TestMsgServer_TransferApplication_Error_DestinationExists(t *testing.T) {
 	foundApp, isSrcFound = k.GetApplication(ctx, srcAddr)
 	require.True(t, isSrcFound)
 	require.Equal(t, srcAddr, foundApp.Address)
-	require.Equal(t, int64(100), foundApp.Stake.Amount.Int64())
+	require.Truef(t,
+		expectedAppStake.Amount.Equal(foundApp.Stake.Amount),
+		"expected %d, got %d",
+		expectedAppStake.Amount.Int64(), foundApp.Stake.Amount.Int64(),
+	)
```
Comment (bryanchriswhite) [inline x/application/keeper/msg_server_transfer_application_stake_test.go:135]:
```suggestion
	require.Equal(t, expectedAppStake.Amount.Int64(), foundApp.Stake.Amount.Int64())
```

File: x/application/keeper/msg_server_stake_application_test.go (hunk 2 above)
Comment (red-0ne) [reply x/application/keeper/msg_server_stake_application_test.go:75]: I believe you could assert equality even for the coins themselves IIRC

**Review by red-0ne — APPROVED** [review]
Great addition. LGTM :muscle:

File: x/application/keeper/msg_server_stake_application_test.go (hunk 2 above)
Comment (bryanchriswhite) [reply x/application/keeper/msg_server_stake_application_test.go:75]: I think I got hung up on a pointer and non-pointer not satisfying `Equal()` and just didn't read the error closely enough. :sweat_smile:

**Pull request merged**